integration/test/performancetest/performance_test.go

```diff
@@ -42,18 +55,58 @@ func TestPerformance(t *testing.T) {
 		t.Fatalf("Error: %v", err)
 	}
 
-	//------Placeholder to put data into database------//
-	//useless code so data get used and compiler isn't mad
 	if data == nil {
 		t.Fatalf("No data")
 	}
-	//data base 
-	dynamoDB := InitializeTransmitterAPI(DynamoDBDataBase) //add cwa version here
-	if dynamoDB == nil{
-		t.Fatalf("Error: generating dynamo table")
-	}
+	
 	_, err = dynamoDB.SendItem(data)
-	if err !=nil{
+	if err != nil {
 		t.Fatalf("Error: couldnt upload metric data to table")
 	}
-}
\ No newline at end of file
+}
+
+func startLogWrite(t *testing.T, filePath1, filePath2 string, agentRunDuration time.Duration, tps int) {
+	//create wait group so main test thread waits for log writing to finish before stopping agent and collecting data
+	var logWaitGroup sync.WaitGroup
+	logWaitGroup.Add(2)
+
+	//start goroutines to write to log files concurrently
+	go func() {
+		defer logWaitGroup.Done()
+		writeToLogs(t, filePath1, agentRunDuration, tps)
+	}()
+	go func() {
+		defer logWaitGroup.Done()
+		writeToLogs(t, filePath2, agentRunDuration, tps)
+	}()
+
+	//wait until writing to logs finishes
+	logWaitGroup.Wait()
+}
+
+func writeToLogs(t *testing.T, filePath string, durationMinutes time.Duration, tps int) {
+	f, err := os.Create(filePath)
+	if err != nil {
+		t.Fatalf("Error occurred creating log file for writing: %v", err)
+	}
+	defer f.Close()
+	defer os.Remove(filePath)
+
+	log.Printf("Writing lines to %s with %d transactions per second", filePath, tps)
+
+	startTime := time.Now()
+
+	//loop until the test duration is reached
+	for currTime := startTime; currTime.Sub(startTime) < durationMinutes; currTime = time.Now() {
+
+		//assume this for loop runs instantly for purposes of simple throughput calculation
+		for i := 0; i < tps; i++ {
+			_, err = f.WriteString(fmt.Sprintf("%s - #%d This is a log line.\n", currTime.Format(time.StampMilli), i))
+			if err != nil {
+				t.Logf("Error occurred writing log line: %v", err)
+			}
+		}
+
+		time.Sleep(1 * time.Second)
+	}
```

Comment:
Part of me wants to rewrite the outer for loop here.. but I'll have to think about it. What I kind of want to do is to change it to a `for-select`, so like:
```go
for {
    case <-time.After(durationMinutes * time.Minute): 
         return
}
```
but don't know if the case where we write is cleaner that way or not.

Reply:
Implemented